demo/src/screens/HomeScreen/HomeCarousel/HomeCarousel.js

```diff
@@ -1,16 +1,106 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 import {Carousel, Image} from 'react-bootstrap';
 import {Link} from 'react-router-dom';
 import slideOne from '../../../images/slide_one.jpg';
 import slideTwo from '../../../images/slide_two.jpg';
 import slideThree from '../../../images/slide_three.jpg';
+import {event} from '../../../lib/gtag.js';
+import {CodeModal} from '../../../components/CodeModal/CodeModal.js';
+import {getItemParameters, getEventCodeSnippet, getMeasureCodeSnippet} from '../../../utils.js';
 import './HomeCarousel.css';
 
 /**
  * Creates slideshow component for showcasing our latest deals and offerings.
- * @return {!JSX} The component.
+ * @return {!JSX}
  */
 export function HomeCarousel() {
+  const viewPromotionParameters = {
+    items: [
+      {
+        ...getItemParameters('7ba94'),
+        promotion_name: 'Home Carousel',
+      },
+      {
+        ...getItemParameters('hjdf7'),
+        promotion_name: 'Home Carousel',
+        discount: 25,
+      },
+      {
+        ...getItemParameters('3h488'),
+        promotion_name: 'Home Carousel',
+      },
+    ],
+    promotion_name: 'Home Carousel',
```

Comment:
It may be outdated / a case of incorrect reading of the docs. I used the event reference to select parameters and promotion_name is both at the top level and item level:

https://developers.google.com/gtagjs/reference/aw-events#view_promotion